cuda_core/cuda/core/experimental/_memory/_buffer.pyx

```diff
@@ -203,58 +209,75 @@ cdef class Buffer:
                 s
             ))
 
-    def fill(self, value: int, width: int, *, stream: Stream | GraphBuilder):
-        """Fill this buffer with a value pattern asynchronously on the given stream.
+    def fill(self, value: int | BufferProtocol, *, stream: Stream | GraphBuilder):
+        """Fill this buffer with a repeating byte pattern.
 
         Parameters
         ----------
-        value : int
-            Integer value to fill the buffer with
-        width : int
-            Width in bytes for each element (must be 1, 2, or 4)
+        value : int | :obj:`collections.abc.Buffer`
+            - int: Must be in range [0, 256). Converted to 1 byte.
```

Comment:
Forgot to ask yesterday:
1. if we can copy up to 4 bytes, why do we restrict ourselves to 1 byte only?
2. why is the value required to be unsigned?